### DISCUSSION TITLE
Possibility of WhiteWind lexicon support?

### DISCUSSION THREAD

**Discussion by ygg2** [bluesky-social/atproto#2501] [Bluesky Client App]
A lot of people expect their whtwnd posts to crosspost to bsky. From what I understand so far, the reason they don't appear naturally is that WhiteWind uses a separate lexicon, and support for other lexicons is not required (which makes sense) or even necessarily recommended. However I think some type of support could show the potential, since it's one of the first other ATproto platforms, and the service it provides is a similar format to Bluesky.

I'm aware Bluesky users have said they don't want to see long text appearing on their feeds, but would it be possible to have WhiteWind posts display as something like preview text + a link, or an embed (but still having the same likes/reposts)?

**Comment by bnewbold** [comment]
We are open to it! Definitely would like a better way to have both WhiteWind specifically, and other independent atproto content be visible as atproto entities in the Bluesky app, and be able to do things like re-share and "like" in a meaningful way.

An interesting comparison is what Bluemotion does for video (described in https://fediversereport.com/video-audio-and-blogging-japanese-bluesky-is-building-in-the-atmosphere/), each video gets a canonical bsky post, and likes/replies/re-shares to that specific post in Bluesky get passed back to Bluemotion. In that case, the obvious extension or native support would be video streaming in-app, similar to what Bluesky does for youtube embeds already.

**Reply by ygg2** [reply]
Thank you! Glad to hear it's planned even if not being worked on yet. I'll look forward to it!

Interesting to see how Bluemotion is handled although I admit I'm a little confused how it manages to crosspost (does it just send the post twice once as its own and once as bsky...?)

**Reply by antont** [reply]
What's the status here now? Somehow WhiteWind at least shows the likes to a Bluesky post where the blog post is shared, but it's maybe different from that canonical post mechanism .. or is it?

**Reply by ygg2** [reply]
@antont Right now as you said the stats are collected from the number of app.bsky.feed.like on app.bsky.feed.post that contain WhiteWind links (it also listens for some other record types that could contain a link like Frontpage). However the shared WhiteWind links are the same as any other link, it's not like the Bluesky post contains a field with information about the record the linked page displays, so in fact if a link was shared to an unknown different frontend, WhiteWind wouldn't know to pick up that it was a link to a com.whtwnd.blog.entry record.

Some other clients have a link to view people's account on WhiteWind, and load com.whtwnd.blog.entry records under a tab which seems like a nice presentation for the viewing sense. For the interacting sense, a solution might be related to saving AT URI in a field to denote the record that's being shared so different applications can check what record the Bluesky post connects to. I think the Frontpage team were also talking about ways to do something like that since their app focuses on link sharing, so fingers crossed that will manifest soon.
